cmd/boots/main.go

```diff
@@ -294,6 +298,33 @@ func getReporter(l log.Logger) (client.Reporter, error) {
 	}
 }
 
+// parseDynamicIPXEVars will parse any number of variable definitions from a
+// string, and return a an array of two-element arrays which are the key/value
+// string pairs of the variable's name and value. These will later be injected
+// as variable definitions in the iPXE script. Variable order is preserved.
+func parseDynamicIPXEVars(v string) [][]string {
+	if v == "" {
+		return nil
+	}
+
+	// extract multiple variable definitions by splitting first on spaces
+	allDefs := strings.Fields(v)
+	retVal := make([][]string, len(allDefs))
+
+	// iterate over each variable definition and split on '='
+	for i, el := range allDefs {
+		varDef := strings.SplitN(el, "=", 2)
+		if len(varDef) == 2 {
+			retVal[i] = varDef
+		} else {
+			err := errors.Errorf("unable to parse IPXE dynamic var definitions from string %v", v)
+			panic(err)
```

Comment:
What do you think about returning an error from this function and letting that propagate up so that `func main` can handle the panic?

Reply:
Yeah, that seems reasonable. This should now be implemented.